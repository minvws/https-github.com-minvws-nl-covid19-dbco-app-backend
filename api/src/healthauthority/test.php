<?php
$clientKeyPair = sodium_crypto_kx_keypair();
echo base64_encode($clientKeyPair);
die;
$clientPublicKey = sodium_crypto_kx_publickey($clientKeyPair);

$haKeyPair = sodium_crypto_kx_keypair();
$haPublicKey = sodium_crypto_kx_publickey($haKeyPair);

$clientSessionKeys = sodium_crypto_kx_client_session_keys($clientKeyPair, $haPublicKey);
$haSessionKeys = sodium_crypto_kx_server_session_keys($haKeyPair, $clientPublicKey);

var_dump($clientSessionKeys);
var_dump($haSessionKeys);

echo base64_encode($clientSessionKeys) . "\n";
echo base64_encode($haSessionKeys) . "\n";

echo ($clientSessionKeys == $haSessionKeys ? "" : "NOT ") . "EQUAL\n";
echo "\n";
