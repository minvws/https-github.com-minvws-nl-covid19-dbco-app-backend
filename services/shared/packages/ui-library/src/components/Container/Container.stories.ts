import type { Meta, StoryFn } from '@storybook/vue';
import Container from './Container.vue';
import { InfoLabel } from '../../../docs/components';

const asValues = ['div', 'span', 'section', 'article'] as const;
const sizeValues = ['sm', 'md', 'lg', 'xl'] as const;

interface Props {
    content: string;
    as: typeof asValues;
    size: typeof sizeValues;
    centeredContent: boolean;
}

const story: Meta = {
    title: 'Components/Container',
    component: Container,
    args: {
        content:
            'Laudantium dolores libero cum ea quae pariatur possimus quo adipisci sit corrupti. Quidem deleniti dolor quo esse facere adipisci assumenda sint praesentium eveniet repudiandae eveniet et iusto. Facere minus unde odit officiis non suscipit optio tenetur quas voluptate pariatur. Voluptate harum excepturi quos veniam excepturi aperiam alias exercitationem commodi aut atque facere quam. Natus rerum modi repellendus debitis cumque maiores in quisquam occaecati in perferendis.',
    },
    argTypes: {
        content: {
            control: 'text',
        },
        as: { options: asValues, control: { type: 'select' } },
        size: { options: sizeValues, control: { type: 'select' } },
        centeredContent: { control: 'boolean' },
    },
};

const Template: StoryFn<Props> = ({ content, ...args }) => {
    return {
        components: { Container },
        setup() {
            return { args, content };
        },
        template: `<Container v-bind="args">{{content}}</Container>`,
    };
};

const SizesTemplate: StoryFn<Props> = ({ content, ...args }) => {
    return {
        components: { Container, InfoLabel },
        setup() {
            return { args, content, sizeValues };
        },
        template: `
        <div>
            <div v-for="size in sizeValues" class='tw-mb-4'>
                <Container v-bind="args" :size='size'>{{content}}</Container>
                <InfoLabel class="tw-mt-2">
                    {{ size }}
                </InfoLabel>
            </div>
        </div>`,
    };
};

export const Default = Template.bind({});

export const Sizes = SizesTemplate.bind({});

export default story;
