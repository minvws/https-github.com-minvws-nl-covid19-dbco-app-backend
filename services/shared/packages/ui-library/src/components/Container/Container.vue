<template>
    <component v-tw-merge :is="as" :class="styles">
        <slot></slot>
    </component>
</template>

<script lang="ts">
import type { PropType } from 'vue';
import { defineComponent } from 'vue';
import type { Extends, Sizes } from '../../types';

type Tag = 'div' | 'span' | 'section' | 'article';
type Size = Extends<Sizes, 'sm' | 'md' | 'lg' | 'xl'>;

const Styles: Record<Size, string[]> = {
    sm: ['tw-max-w-lg'],
    md: ['tw-max-w-3xl'],
    lg: ['tw-max-w-5xl'],
    xl: ['tw-max-w-[1440px]'],
};

export default defineComponent({
    props: {
        as: { type: String as PropType<Tag>, default: 'div' },
        size: { type: String as PropType<Size>, default: 'xl' },
        centerContent: Boolean,
    },
    setup({ as, size, centerContent }) {
        const styles = ['tw-block', 'tw-w-full', 'tw-px-4', 'tw-mx-auto', ...Styles[size]];

        if (centerContent) {
            styles.push('tw-flex', 'tw-flex-col', 'tw-items-center');
        }

        return {
            as,
            styles,
        };
    },
});
</script>
