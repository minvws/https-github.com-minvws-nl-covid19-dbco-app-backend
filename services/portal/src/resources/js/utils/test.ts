import Modal from '@/components/plugins/modal';
import safeHtml from '@/directives/safeHtml';
import FiltersPlugin from '@/plugins/filters';
import TypingHelpers from '@/plugins/typings';
import VueFormulate from '@braid/vue-formulate';
import { registerDirectives } from '@dbco/ui-library';
import type { DecoratedWrapper } from '@dbco/ui-library/test';
import { flushCallStack, decorateWrapper } from '@dbco/ui-library/test';
import { faker } from '@faker-js/faker';
import type { RenderResult } from '@testing-library/vue';
import type { Wrapper, WrapperArray } from '@vue/test-utils';
import { createLocalVue } from '@vue/test-utils';
import BootstrapVue from 'bootstrap-vue';
import { PiniaVuePlugin } from 'pinia';
import VueMask from 'v-mask';
import type Vue from 'vue';
import { defineComponent, type ComponentCustomProperties, type VueConstructor } from 'vue';
import VueI18n from 'vue-i18n';
import Vuex from 'vuex';
import { compileToFunctions } from 'vue-template-compiler';

export { flushCallStack, decorateWrapper };

export type UntypedWrapper<T extends Vue = Vue> = Wrapper<T> & {
    // Currently the component types are not available outside the Vue component files.
    // To prevent (wrapper as any) in all the tests we provided this workaround.
    // eslint-disable-next-line @typescript-eslint/no-explicit-any
    vm: Wrapper<Vue>['vm'] & Record<string, any> & { $refs: Record<string, any> };
};

type WrapperOrRender<TWrapper extends Wrapper<Vue>> = TWrapper | RenderResult;

const routerLink = defineComponent({
    setup() {
        return { navigate: vi.fn() };
    },
    render: compileToFunctions('<div><slot href="/" :navigate="navigate" /></div>').render,
});

export function createDefaultLocalVue() {
    const localVue = createLocalVue();

    localVue.mixin({
        created() {
            // when the root element is created we set a global app reference
            if (!(this as Vue).$parent) {
                window.app = this as Vue & ComponentCustomProperties;
            }
        },
    });

    localVue.use(BootstrapVue);
    localVue.use(Modal);
    localVue.use(PiniaVuePlugin);
    localVue.use(Vuex);
    localVue.use(VueI18n);
    localVue.use(VueMask);
    localVue.use(VueFormulate);
    localVue.use(FiltersPlugin);
    localVue.use(TypingHelpers);

    localVue.directive('safe-html', safeHtml);

    localVue.component('router-link', routerLink);

    registerDirectives(localVue);

    return localVue;
}
/**
 * Use this setup function to create your testcomponent factory. It will help with:
 * - destroying the wrapper after the test (to prevent memory leaks)
 * - creating the localVue instance
 *
 * @param factory a function that takes the localvue and any custom arguments and returns a wrapper or render result
 * @returns a function that takes the custom arguments and returns a wrapper or render result. Use this function to create your components
 */
interface ISetupTest {
    <TProperties extends unknown[], TRenderResult extends Promise<RenderResult>>(
        factory: (localVue: VueConstructor, ...args: TProperties) => TRenderResult
    ): (...args: TProperties) => TRenderResult;
    <TProperties extends unknown[], TRenderResult extends RenderResult>(
        factory: (localVue: VueConstructor, ...args: TProperties) => TRenderResult
    ): (...args: TProperties) => TRenderResult;
    <TProperties extends unknown[], TComponent extends Vue, TWrapper extends Wrapper<TComponent>>(
        factory: (localVue: VueConstructor, ...args: TProperties) => TWrapper
    ): (...args: TProperties) => DecoratedWrapper<TWrapper> & UntypedWrapper;
    <TProperties extends unknown[], TComponent extends Vue, TWrapper extends Promise<Wrapper<TComponent>>>(
        factory: (localVue: VueConstructor, ...args: TProperties) => TWrapper
    ): (...args: TProperties) => Promise<DecoratedWrapper<Awaited<TWrapper>> & UntypedWrapper>;
}

let localVue: VueConstructor;
export const setupTest: ISetupTest = (factoryFunction: Parameters<ISetupTest>[0]) => {
    if (localVue) {
        throw new Error('Only one setup allowed per test file');
    }

    localVue = createDefaultLocalVue();
    let wrapperOrRender: WrapperOrRender<Awaited<ReturnType<typeof factoryFunction>>> | null;

    afterEach(() => {
        if (wrapperOrRender) {
            if (isRenderResult(wrapperOrRender)) {
                wrapperOrRender.unmount();
            } else {
                wrapperOrRender.destroy();
            }
        }
        wrapperOrRender = null;
    });

    afterEach(() => {
        vi.clearAllMocks();
    });

    return <TProperties extends unknown[]>(...args: TProperties) => {
        if (wrapperOrRender)
            throw new Error('Cannot create multiple wrappers in a single test case, try to use `it.each` instead');

        const newWrapperOrRender = factoryFunction(localVue, ...args);

        if (newWrapperOrRender instanceof Promise) {
            return newWrapperOrRender.then((resolvedWrapper) => {
                wrapperOrRender = resolvedWrapper;
                return isRenderResult(wrapperOrRender) ? wrapperOrRender : decorateWrapper(wrapperOrRender);
            });
        }

        wrapperOrRender = newWrapperOrRender;
        return isRenderResult(wrapperOrRender) ? wrapperOrRender : decorateWrapper(wrapperOrRender);
    };
};

function isRenderResult<TWrapper extends Wrapper<Vue>>(result: WrapperOrRender<TWrapper>): result is RenderResult {
    return !!(result as RenderResult)?.isUnmounted;
}

/**
 * Decorator for faker.
 * @returns faker, decorated with custom helpers.
 */
export const fakerjs = {
    ...faker,
    custom: {
        arrayOfUuids: (min = 1, max = 10) =>
            faker.helpers.uniqueArray(faker.string.uuid, faker.number.int({ min, max })),
        typedArray: <Type>(el: Type, min = 1, max = 10) => [...Array(faker.number.int({ min, max }))].map(() => el),
    },
};

/**
 * Creates a container in the document body.
 * @param {string} [tag=div] - The tag of the container
 * @returns {HTMLElement} The newly created container
 */
export const createContainer = (tag = 'div') => {
    const container = document.createElement(tag);
    document.body.appendChild(container);
    return container;
};

export function waitForElements(wrapper: Wrapper<Vue>, selector: string, timeout = 500): Promise<WrapperArray<Vue>> {
    return new Promise((resolve, reject) => {
        const start = Date.now();
        const interval = setInterval(() => {
            const elements = wrapper.findAll(selector);
            if (!elements.length) {
                if (Date.now() - start > timeout) {
                    clearInterval(interval);
                    reject(`Timeout of ${timeout}ms reached while looking for element ${selector}`);
                }

                return;
            }

            clearInterval(interval);
            resolve(elements);
        }, 10);
    });
}
