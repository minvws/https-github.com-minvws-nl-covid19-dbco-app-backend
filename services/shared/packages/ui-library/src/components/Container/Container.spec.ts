import { shallowMount } from '@vue/test-utils';
import { faker } from '@faker-js/faker';
import Container from './Container.vue';
import { createDefaultLocalVue } from '../../test/local-vue';

function createComponent(propsData: object, content?: string) {
    return shallowMount(Container, {
        localVue: createDefaultLocalVue(),
        propsData,
        slots: {
            default: content || faker.lorem.sentence(),
        },
    });
}

describe('Container.vue', () => {
    it('should be able to render with default props', () => {
        const wrapper = createComponent({}, 'test content');
        expect(wrapper.html()).toMatchSnapshot();
    });

    it('should be able to render with size prop', () => {
        const wrapper = createComponent({ size: 'md' });
        expect(wrapper.classes()).include('tw-max-w-3xl');
    });

    it('should be able to render with centered content', () => {
        const wrapper = createComponent({ centerContent: true });
        expect(wrapper.classes()).toEqual(expect.arrayContaining(['tw-flex', 'tw-flex-col', 'tw-items-center']));
    });
});
