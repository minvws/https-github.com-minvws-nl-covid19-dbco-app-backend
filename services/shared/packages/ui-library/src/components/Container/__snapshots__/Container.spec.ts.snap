// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Container.vue > should be able to render with default props 1`] = `"<div class=\\"tw-block tw-w-full tw-px-4 tw-mx-auto tw-max-w-[1440px]\\">test content</div>"`;
